React 룰 추가

antiline opened the pull request:
React 관련된 기본 룰만 추가 했습니다.
계정팀 내부에서 사용하는 느슨한 룰은 추가 하지 않았습니다.

Commit:
Add react options

Commit:
Remove comma

Commit:
누락된 상수 추가

namenu commented on index.js:
```diff
@@ -1,5 +1,15 @@
+const OFF = 0;
```
0, 1, 2 대신 "off", "warn", "error" 라고도 표기할 수 있습니다.
https://eslint.org/docs/user-guide/getting-started#configuration

namenu commented on index.js:
```diff
@@ -1,5 +1,15 @@
+
 module.exports = {
   extends: 'airbnb',
+  'globals': {
```
`React`, `ReactDOM` 둘 다 명시적으로 import 하고 사용하는게 바람직해 보입니다.

namenu commented on index.js:
```diff
@@ -1,5 +1,15 @@
+  },
+  'plugins': ['react'],
   rules: {
+    'react/jsx-filename-extension': [ON, { 'extensions': ['.jsx'] }],
```
이 규칙을 포함하는 이유는 무엇인가요?

antiline replied:
UI 관련된 JSX 문법이 여러 파일에 난잡하게 위치하지않게 하기 위해 JSX 는 jsx 확장자 파일에만 넣도록 강제하는 룰 입니다.
React 특성상 완벽하게 MVC가 되지 않지만 그래도 V 코드를 한쪽에만 정리 할 수 있을 거라 생각해서 팀 내부에서는 사용하고 있습니다.
관련해서 전사적으로 공유할만한 룰이라 생각되서 추가했습니다.

Commit:
eslint 수정

Commit:
'off' 명시적 사용으로 수정

Commit:
ON -> 'warn' 명시적 사용으로 수정

Commit:
`jsx-a11y/label-has-for` 규칙 변경

- 디폴트 설정은 `"required": { "every": ["nesting", "id"] }`로 `<label>` 태그는 항상 컨트롤을 포함하고 각각 `id`와 `htmlFor` 속성을 가져야 함
- `<label>` 태그가 컨트롤을 포함하는 것이 강제될 경우 RUI의 마크업 방식에 어긋남
- 따라서 `id`와 `htmlFor` 속성만 가지면 되도록 규칙을 완화

kyungmi commented:
`jsx-a11y/label-has-for` 관련 규칙도 수정해 보았습니다. 자세한 내용은 commit 메세지를 확인해주세요.

kyungmi commented on index.js:
```diff
@@ -1,5 +1,16 @@
+  globals: {
+    React: true,
+    ReactDOM: true,
+  },
```
global 변수에 `React`와 `ReactDOM`이 포함되어야 하는 이유는 무엇인가요?

kyungmi commented on index.js:
```diff
@@ -1,5 +1,16 @@
+    React: true,
+    ReactDOM: true,
+  },
+  plugins: ['react'],
```
`react` plugin을 명시적으로 써주신 이유가 있나요? `jsx-a11y` 도 함께 써주는게 좋을까요?

kyungmi commented on index.js:
```diff
@@ -1,5 +1,16 @@
+  plugins: ['react'],
   rules: {
+    // JSX 파일의 확장자를 `.jsx`로 제한
+    'react/jsx-filename-extension': ['warn', { extensions: ['.jsx'] }],
```
`react/jsx-filename-extension`은 이미 airbnb rule에 포함되어 있는 내용인데 `error`에서 `warn`으로 레벨을 낮추신 이유가 궁금하네요.

Commit:
airbnb 가이드에 포함된 react 관련 룰 제거

antiline commented:
airbnb 에 포함되어 있고, 느슨한거 빼고, 잘못 정리된것 빼니 `jsx-a11y/label-has-for`만 남는군요.
@namenu 해당 룰만 추가하고 정리하면 될것 같습니다.
